Assets/Scripts/PlayerController.cs

```diff
@@ -21,23 +35,44 @@ public class PlayerController : MonoBehaviour
     private GameObject shot;
     [SerializeField]
     private Transform shotSpawn;
+    [SerializeField]
+    private AudioClip shotSFX;
+    [SerializeField]
+    private GameObject bomb;
+    [SerializeField]
+    private float bombCooldown;
+    [SerializeField]
+    private AudioClip bombSFX;
+    [SerializeField]
+    private int numBombs;
 
     private new Rigidbody rigidbody;
+    private AudioSource audioSource;
     private float nextFire;
 
     void Start()
     {
         rigidbody = GetComponent<Rigidbody>();
+        audioSource = GetComponent<AudioSource>();
```

Comment:
how are you guaranteeing this won't be null (RequireComponent)

Double check with Assert.IsNotNull

Reply:
You haven't done any null checks here or assert checks for all of these serialized fields you have